src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Storage class to load and save file to local drive.
+ */
+public class Storage {
+
+    private final String filePath;
+
+    /**
+     * Construct a Storage object to load and save file to local drive.
+     *
+     * @param filePath  Filepath of where to load/save file from/to.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Load file and read data line by line.
+     * Convert the data to Task objects.
+     * Return an ArrayList containing the Task objects.
+     *
+     * @return  ArrayList containing the Task objects.
+     * @throws IOException  If fail to read the file.
+     */
+    public ArrayList<Task> load() throws IOException {
+        ArrayList<Task> loadedTasks = new ArrayList<>();
+        FileReader fReader = new FileReader(String.valueOf(this.filePath));
+        BufferedReader reader = new BufferedReader(fReader);
+        String line;
+        while ((line = reader.readLine()) != null) {
+            String[] strArr;
+            Task task = new Task("");
+            strArr = line.split(" \\| ");
+            switch (strArr[0]) {
+            case "T":
+                task = new Todo(strArr[2]);
+                if (strArr[1].equals("1")) {
```

Comment:
```suggestion
                if (strArr[1].equals("X")) {
```
I think there is a bug in which marked to-do tasks are being loaded as unmarked to-do tasks upon re-running Duke (which might be caused by this line)

Reply:
You are right! Thanks so much for noticing this! Fixed the issue in latest commit.